server/enroll.go

```diff
@@ -157,37 +157,29 @@ func (c *context) sign(keyId string, principals []string, serial uint64, certTyp
 // This assumes there's an authenticating proxy which provides the user in a header, configurable.
 // We identify the proxy with its TLS client cert
 // TODO: this function needs test coverage
-func proxyAuthenticated(ap *config.AuthenticatingProxy, requestedUser string, w http.ResponseWriter, r *http.Request) bool {
+func proxyAuthenticated(ap *config.AuthenticatingProxy, userAuthed string, w http.ResponseWriter, r *http.Request) bool {
```

Comment:
I think this function should be changed to return the user, rather than reading the header in the caller & passing it in

Reply:
That ensures all the proxy handling code (mtls requirement + header handling) are in one single atomic testable unit

Reply:
I've rewritten this a bit and added more tests. How's that now?

Reply:
looks great now, thank you!